visualization/generate_experimentId.py

```diff
@@ -0,0 +1,86 @@
+import pandas as pd
+import numpy as np
+import numbers
+
+
+def generate_experimentId(measurement_data):
+    '''
+    automatically generate the experimentId:
+    every row which shares the observableParameters (if provided),
+    noiseParameters and observableTransformation gets
+    the same experimentId, for the assignment all numeric values in the
+    noiseParameters are considered to be the same
+
+    Parameters:
+    ----------
+
+    measurement_data: pandas data frame for measurement data
+
+    Return:
+    ----------
+
+    measurement_data: pandas data frame for measurement data with column for
+    experimentId added
+
+    '''
+    observableParameters = np.array(measurement_data.observableParameters)
+    noiseParameters = np.array(measurement_data.noiseParameters)
+    observableTransformation = np.array(
+        measurement_data.observableTransformation
+    )
+
+    # check if there occurs the name 'nan' in the observableParameter column
+    # , should not occur!
+    checkNaming = np.where(observableParameters == 'empty')
```

Comment:
are you checking for 'nan' or 'empty'?

Reply:
Maybe also check if the 'expermentId' column already exists? Then I can always run this `generate_experimentId` after importing `measurment_data`.